### PR TITLE
Geochemistry: constants added, robust activity, swapper enhancements

### PR DESCRIPTION
**Pull request by WilkAndy**
You've seen the Constants stuff, @loganharbour .   Here are a few extra additions (not too many lines - will hopefully take you only about 5 mins to review):

- GeochemistryActivity uses the Constants, and also is made a bit more robust against unphysical inputs

- The Swapper is enhanced so it computes bulk composition in the new basis, which is needed when actual simulations are performed

### CODE REVIEW

**Comment by WilkAndy** [comment]
The failure is `FAIL dampers/min_damping.min_nodal_damping FAILED (CRASH) [min_threads=2]` which is unrelated to this PR.

**Comment by WilkAndy** [comment]
Feel like some reviewing, @loganharbour ?

File: modules/geochemistry/include/utils/GeochemistryConstants.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+#pragma once
+
+#include "MooseTypes.h"
+
+constexpr Real MOLES_PER_KG_WATER = 55.51;
+constexpr Real LOGTEN = 2.30258509299404;
+constexpr Real FARADAY = 96485.3415;    // Coulombs . mol^-1
+constexpr Real GAS_CONSTANT = 8.314472; // m^2 . kg . s^-2 . K^-1 . mol^-1
+constexpr Real CELSIUS_TO_KELVIN = 273.15;
```
Comment (loganharbour) [inline modules/geochemistry/include/utils/GeochemistryConstants.h:18]: This should all be in a namespace. 

Also: you're sneaky here putting in constants you haven't used yet 😬

**Review by loganharbour — CHANGES_REQUESTED** [review]
Minor request. Looks good.

File: modules/geochemistry/include/utils/GeochemistryConstants.h (hunk 1 above)
Comment (WilkAndy) [reply modules/geochemistry/include/utils/GeochemistryConstants.h:18]: Heheheh, yea, those constants are used in my non-pushed stuff.  Hopefully you don't mind.  I'll make the namespace in  few mins.

**Review by loganharbour — APPROVED** [review]

**Comment by WilkAndy** [comment]
Woohoo, thanks for the review and merge, @loganharbour

**Pull request merged**